falcon/asgi/request.py

```diff
@@ -716,6 +733,10 @@ async def get_media(self):
                 self.content_type,
                 self.content_length
             )
+        except Exception:
+            if when_empty_fallback is _UNSET or self.stream.tell() != 0:
```

Comment:
Is there a test for ASGI without the `Content-Length` header? It might be that our test framework currently even does not offer good support for that...

But it would be good to add tests for failing media deserialization, where the `Content-Length` header is unset, both when there is a streamed body, and when not.

Reply:
I'm not sure, but I think tell returns the number of read bytes, regardless of it the body was streamed or not. After trying to de-serialize the media it should be `> 0` if there was a body.

I can look into trying to omit content-length